compact.cpp

```diff
@@ -32,7 +32,7 @@ unsigned __stdcall ObjectMemory::lastOTEntry()
 	const OTE* pOT = m_pOT;
 	while (pOT[i].isFree())
 	{
-		ASSERT(i >= 0);
+		ASSERT(0 < i);
```

Comment:
Yes, good point. I was more reacting to the inverted condition. I know this style is of having the constant on the lhs is recommended by some (mainly because it avoids the problem of accidentally using assignment where an equality comparison was intended), however it is:

- harder to read
- not consistent with the general style of the VM code
- perhaps better achieved using the compiler warning that detects assignment in a condition expression.
In this case, the assertion is a bit pointless, so might as well be removed.